test/e2e/storage/vsphere/config.go

```diff
@@ -0,0 +1,245 @@
+		// Datastore in which vmdks are stored.
+		// Deprecated. See Workspace.DefaultDatastore
+		DefaultDatastore string `gcfg:"datastore"`
+		// WorkingDir is path where VMs can be found. Also used to create dummy VMs.
+		// Deprecated.
+		WorkingDir string `gcfg:"working-dir"`
+		// Soap round tripper count (retries = RoundTripper - 1)
+		RoundTripperCount uint `gcfg:"soap-roundtrip-count"`
+		// Deprecated as the virtual machines will be automatically discovered.
+		// VMUUID is the VM Instance UUID of virtual machine which can be retrieved from instanceUuid
+		// property in VmConfigInfo, or also set as vc.uuid in VMX file.
+		// If not set, will be fetched from the machine via sysfs (requires root)
+		VMUUID string `gcfg:"vm-uuid"`
+		// Deprecated as virtual machine will be automatically discovered.
+		// VMName is the VM name of virtual machine
+		// Combining the WorkingDir and VMName can form a unique InstanceID.
+		// When vm-name is set, no username/password is required on worker nodes.
+		VMName string `gcfg:"vm-name"`
+	}
+
+	VirtualCenter map[string]*Config
+
+	Network struct {
+		// PublicNetwork is name of the network the VMs are joined to.
+		PublicNetwork string `gcfg:"public-network"`
+	}
+
+	Disk struct {
+		// SCSIControllerType defines SCSI controller to be used.
+		SCSIControllerType string `dcfg:"scsicontrollertype"`
+	}
+
+	// Endpoint used to create volumes
+	Workspace struct {
+		VCenterIP        string `gcfg:"server"`
+		Datacenter       string `gcfg:"datacenter"`
+		Folder           string `gcfg:"folder"`
+		DefaultDatastore string `gcfg:"default-datastore"`
+		ResourcePoolPath string `gcfg:"resourcepool-path"`
+	}
+}
+
+// GetVSphereInstances parses vsphere.conf and returns VSphere instances
+func GetVSphereInstances() (map[string]*VSphere, error) {
+	cfg, err := getConfig()
+	if err != nil {
+		return nil, err
+	}
+	return populateInstanceMap(cfg)
+}
+
+func getConfig() (*ConfigFile, error) {
+	if confFileLocation == "" {
+		return nil, fmt.Errorf("Env variable 'VSPHERE_CONF_FILE' is not set.")
+	}
+	confFile, err := os.Open(confFileLocation)
+	if err != nil {
+		return nil, err
+	}
+	defer confFile.Close()
+	cfg, err := readConfig(confFile)
+	if err != nil {
+		return nil, err
+	}
+	return &cfg, nil
+}
+
+// readConfig parses vSphere cloud config file into ConfigFile.
+func readConfig(config io.Reader) (ConfigFile, error) {
+	if config == nil {
+		err := fmt.Errorf("no vSphere cloud provider config file given")
+		return ConfigFile{}, err
+	}
+
+	var cfg ConfigFile
+	err := gcfg.ReadInto(&cfg, config)
+	return cfg, err
+}
+
+func populateInstanceMap(cfg *ConfigFile) (map[string]*VSphere, error) {
+	vsphereInstances := make(map[string]*VSphere)
+
+	// Check if the vsphere.conf is in old format. In this
+	// format the cfg.VirtualCenter will be nil or empty.
+	if cfg.VirtualCenter == nil || len(cfg.VirtualCenter) == 0 {
+		framework.Logf("Config is not per vSphere and is in old format")
```

Comment:
I think we should support only new config format.

Reply:
+1. Since this is for tests, we need not worry about older format.

Reply:
Done.